### PR TITLE
[SPARK-42002][CONNECT][PYTHON] Implement DataFrameWriterV2

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
Implement DataFrameWriterV2

### Why are the changes needed?
Parity with PySpark

### Does this PR introduce _any_ user-facing change?
Yes, implements DataFrameWriterV2

### How was this patch tested?
New UTs

### CODE REVIEW

**Comment by techaddict** [comment]
@HyukjinKwon @zhengruifeng can you review this, thanks 🙏🏼

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -1413,6 +1413,95 @@ def _repr_html_(self) -> str:
...
+        return (
+            f"<ul><li>WriteOperationV2 <br />table_name='{self.table_name}'<br />"
+            f"provider: '{self.provider}' <br />"
+            f"partitioning_columns: '{self.partitioning_columns}' <br />"
+            f"options: '{self.options}' <br />"
+            f"table_properties: '{self.table_properties}' <br />"
+            f"mode: '{self.mode}' <br />"
+            f"</li></ul>"
+        )
+        pass
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/plan.py:1502]:
```suggestion
```

File: python/pyspark/sql/connect/plan.py [hunk 2]
```diff
@@ -1413,6 +1413,95 @@ def _repr_html_(self) -> str:
...
+            f"{i}"
+            f"<WriteOperationV2 table_name='{self.table_name}' "
+            f"provider='{self.provider}' "
+            f"partitioning_columns='{self.partitioning_columns}' "
+            f"options='{self.options}' "
+            f"table_properties='{self.table_properties}' "
+            f"mode='{self.mode}'>"
+        )
+
+    def _repr_html_(self) -> str:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/plan.py:1492]: I actually think you can remove `print` and `_repr_html_` implementation (and let the default implementation handle it)

**Review by HyukjinKwon — APPROVED** [review]
Looks pretty good from a cursory look. cc @zhengruifeng, @amaliujia, @grundprinzip and @xinrong-meng  FYI.

**Comment by HyukjinKwon** [comment]
@techaddict mind rebasing this so we can get the latest result of the testing? Sorry it slipped through my fingers.

**Comment by techaddict** [comment]
@HyukjinKwon thanks, updated the PR

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**